otel-tests-macro/src/lib.rs

```diff
@@ -0,0 +1,74 @@
+use proc_macro::TokenStream;
+use quote::quote;
+use quote::ToTokens;
+use syn::ItemFn;
+use syn::Lit;
+use syn::MetaNameValue;
+use syn::Path;
+
+#[proc_macro_attribute]
+pub fn otel_test(attribute: TokenStream, item: TokenStream) -> TokenStream {
+    let fn_item = syn::parse::<ItemFn>(item).unwrap();
+
+    let sig = fn_item.sig;
+    let block = fn_item.block;
+    let name = &sig.ident;
+    let attrs = fn_item.attrs;
+
+    let test = if attrs.is_empty() {
+        quote!(#[otel_tests::tokio::test])
+    } else {
+        quote!()
+    };
+
+    let init_tracing = if attribute.is_empty() {
+        quote!(::otel_tests::init_tracing)
```

Comment:
❓ Is this a bit hacky or a standard way of defining proc macros? We seem to both depend on `otel-tests` here, but we are also a dependency of the `otel-tests` crate.

Reply:
I guess this is just a way to get around the fact that you need a separate crate for proc macros.

Reply:
Indeed, that was the aim here